adding in a geth metrics dashboard

joshuafernandes opened the pull request:
Hi @nmvalera, when you get a chance could you spin up the quorum quickstart and open up localhost:3000, pick the `geth overview` and have a quick look see please? If I've missed some metrics we can add, but figured this will be a good start?

@bgravenorst ^ and we can clean up the docs and remove those steps

nmvalera reviewed (commented):
Generally looks great! I think having the main Overview table then having  focused on some given metrics is very intuitive

# Comments/Questions (not necessarily to be treated now but can be kept as notes)

## General

- Would it make sense to split Quorum and Tessera into 2 different dashboards so possibly we can possibly have the same format with an overview table for Tessera as well?

- Would it makes sense to also have metrics related to the Network/DevP2P 
    - Diagram with the count of peers over time
    - Is there anything available about the devp2p traffic flowing in and out of the node?
- Would it makes sense to also have metrics related to the JSON-RPC API usage?

## About diagrams

### Overview table

- Why do we need two colums _Latest Block_ and _Latest Header_?
- What does the _Latest Receipt_ column stand for?
- Are they other information that could appear on the table?
    - Regular/Validator boolean flag 
    - Other sync progress fields (c.f. https://github.com/ConsenSys/quorum/blob/v21.4.1/interfaces.go#L100)

### Latest block

Looks great.

### Latest Time

- The Y vertical axis seems to be on a logarithmic scale, is this on purpose? If yes could you please elaborate?

# Block processing

- Does Grafana allow to add some guidelines/descriptions of the metrics so a user can refer to it to understand what does each line measure exactly?

### Transaction pool

- Looks good

### Transaction propagation

- Looks good

### Tessera Raw Request rate

- How did you choose the routes to monitor?

# Tessera Party Info Rate

- Would it possible to have a sense of the number of peers that Tessera is connected to?

### System

- Maybe putting it below Database and Trie (so system which is the lower layer comes at the end)

### Database 

Looks good

### Trie Stats

Maybe rename _Trie Stats_ -> _Trie_ for consistency with other diagrams

joshuafernandes commented:
> Generally looks great! I think having the main Overview table then having focused on some given metrics is very intuitive
> 
> # Comments/Questions (not necessarily to be treated now but can be kept as notes)
> ## General
> * Would it make sense to split Quorum and Tessera into 2 different dashboards so possibly we can possibly have the same format with an overview table for Tessera as well?
Yep, this is a good idea. I'll move the tessera metrics to a seperate dash

> * Would it makes sense to also have metrics related to the Network/DevP2P
>   

>   * Diagram with the count of peers over time
>   * Is there anything available about the devp2p traffic flowing in and out of the node?
Done. Addded and moved the network traffic to a new DevP2P row
 
> * Would it makes sense to also have metrics related to the JSON-RPC API usage?
> 
> ## About diagrams
> ### Overview table
> * Why do we need two colums _Latest Block_ and _Latest Header_?
> * What does the _Latest Receipt_ column stand for?
Removed, 
> * Are they other information that could appear on the table?
>   
>   * Regular/Validator boolean flag
>   * Other sync progress fields (c.f. https://github.com/ConsenSys/quorum/blob/v21.4.1/interfaces.go#L100)
> 
There's a few metrics that seem missing.  @nmvalera any idea if quorum complies with https://eips.ethereum.org/EIPS/eip-2159? 
> ### Latest block
> Looks great.
> 
> ### Latest Time
> * The Y vertical axis seems to be on a logarithmic scale, is this on purpose? If yes could you please elaborate?
Fixed, linear now. Had it on log to focus in on the majority but I'll leave the scale to take care of that
> 
> ### Block processing
> * Does Grafana allow to add some guidelines/descriptions of the metrics so a user can refer to it to understand what does each line measure exactly?
We can put something in a description field - I'll ping you when this is merged and we can add something about each type there?
> 
> ### Transaction pool
> * Looks good
> 
> ### Transaction propagation
> * Looks good
> 
> ### Tessera Raw Request rate
> * How did you choose the routes to monitor?
Based on docs here: https://docs.tessera.consensys.net/en/latest/HowTo/Use/Monitoring/#example-dashboard
> 
> ### Tessera Party Info Rate
> * Would it possible to have a sense of the number of peers that Tessera is connected to? 
I'll put this in a separate dashboard 
> 
> ### System
> * Maybe putting it below Database and Trie (so system which is the lower layer comes at the end)
Done
> 
> ### Database
> * Looks good
> 
> ### Trie Stats
> * Maybe rename _Trie Stats_ -> _Trie_ for consistency with other diagrams

Done
